File: SQL/Select/SQLSelectGroupByField.vb
' ___________________________________________________
'
'  (c) Hi-Integrity Systems 2010. All rights reserved.
'  www.hisystems.com.au<http://www.hisystems.com.au> - Toby Wicks
' ___________________________________________________
'

Option Strict On
Option Explicit On

Namespace SQL

    Public Class SQLSelectGroupByField

        Private pobjExpression As SQLExpression

        Friend Sub New(ByVal objExpression As SQLExpression)

            If objExpression Is Nothing Then
                Throw New ArgumentNullException
            End If

            pobjExpression = objExpression

        End Sub

        Friend ReadOnly Property Expression As SQLExpression
            Get

                Return pobjExpression

            End Get
        End Property

        Friend ReadOnly Property SQL(ByVal eConnectionType As Database.ConnectionType) As String
            Get

                Return pobjExpression.SQL(eConnectionType)

            End Get
        End Property

    End Class

End Namespace

File: SQL/Select/SQLSelectGroupByField.cs
// ___________________________________________________
//
//  (c) Hi-Integrity Systems 2010. All rights reserved.
//  www.hisystems.com.au<http://www.hisystems.com.au> - Toby Wicks
// ___________________________________________________
//

using System.Collections;
using System;
using System.Data;

namespace DatabaseObjects.SQL
{
	public class SQLSelectGroupByField
	{
		private SQLExpression pobjExpression;
			
		internal SQLSelectGroupByField(SQLExpression objExpression)
		{
			if (objExpression == null)
				throw new ArgumentNullException();
				
			pobjExpression = objExpression;
		}
			
		internal SQLExpression Expression
		{
			get
			{
				return pobjExpression;
			}
		}
			
		internal string SQL(Database.ConnectionType eConnectionType)
		{
			return pobjExpression.SQL(eConnectionType);
		}
	}
}
